pkgs/development/compilers/hop/default.nix

```diff
@@ -1,21 +1,21 @@
 { stdenv, fetchurl, bigloo }:
 
 stdenv.mkDerivation rec {
-  name = "hop-2.5.1";
+  name = "hop-3.3.0";
   src = fetchurl {
     url = "ftp://ftp-sop.inria.fr/indes/fp/Hop/${name}.tar.gz";
-    sha256 = "1bvp7pc71bln5yvfj87s8750c6l53wjl6f8m12v62q9926adhwys";
+    sha256 = "14gf9ihmw95zdnxsqhn5jymfivpfq5cg9v0y7yjd5i7c787dncp5";
   };
 
-  buildInputs = [ bigloo ];
-
-  preConfigure = ''
-    export NIX_LDFLAGS="$NIX_LDFLAGS -lbigloogc-4.1a";
+  postPatch = ''
+    substituteInPlace configure --replace "(os-tmp)" '(getenv "TMPDIR")'
   '';
 
+  buildInputs = [ bigloo ];
+
   configureFlags = [
     "--bigloo=${bigloo}/bin/bigloo"
-    "--bigloolibdir=${bigloo}/lib/bigloo/4.1a/"
+    "--bigloolibdir=${bigloo}/lib/bigloo/4.3h/"
```

Comment:
You might be able to lower the future maintenance burden here by using `builtins.parseDrvName` and `bigloo.name` to extract the correct version.

Reply:
Parsing a name in order to guess an install directory seems fishy. Maintaining fishy code may not qualify for lowering the maintenance burden.

Thanks nonetheless for the suggestion.

Reply:
Dude, extracting the version number from a package's `name` is the standard way of doing it (that's why there's a `parseDrvName` function in `builtins`) and we "guess" directory names from version numbers every time we use `python.sitePackages` (everywhere) - but you're the maintainer...